src/module.c

```diff
@@ -9357,11 +9380,10 @@ void moduleFireServerEvent(uint64_t eid, int subid, void *data) {
 
             /* Event specific context and data pointer setup. */
             if (eid == REDISMODULE_EVENT_CLIENT_CHANGE) {
-                modulePopulateClientInfoStructure(&civ1,data,
-                                                  el->event.dataver);
+                serverAssert(modulePopulateClientInfoStructure(&civ1,data, el->event.dataver) == REDISMODULE_OK);
```

Comment:
Isn't it better to push this check earlier to `RM_SubscribeToServerEvent()` and return an error if the version is not supported? We can avoid unexpected assertions and also let a module take some fallback strategy.

Reply:
IIUC he did add a check with graceful failure in RM_SubscribeToServerEvent.
this assertion is just a safeguard.

p.s. note that there are quite a few other places that populate an event structure like this one.
the difference is that others don't have a special function for that work, so the assertion was only added for these two.

examples:
1. RedisModuleFlushInfoV1 populated in this same function (moduleFireServerEvent)
2. RedisModuleSwapDbInfo populated in db.c
3. RedisModuleCronLoopV1 populated in server.c